### ISSUE TITLE
Does it support Python as data source or python in the data transformation ?

### ISSUE THREAD

**Issue by akj2784** [otykier/TabularEditor#590]
**What are you trying to achieve?**
Provide a brief background description of the problem you're trying to solve. Include information about the semantic engine you're working on (Power BI Desktop, Azure Analysis Services, etc.), as not everything is possible across engines.

**What have you tried so far?**
If applicable, describe the steps you have tried so far. For scripting-related questions, show the code that you have written.

**Comment by otykier** [comment]
I’m not sure how this question is related to Tabular Editor.

**Comment by akj2784** [comment]
We are trying to check if we can replace the SSDT tool with Tabular editor for our development work. And SSDT has lots of connectors but it doesn't support Python yet. So was curious to know about tabular editor.

**Comment by otykier** [comment]
Are you talking about [Python in the Query Editor](https://docs.microsoft.com/en-us/power-bi/connect-data/desktop-python-in-query-editor)?

So first of all, Tabular Editor doesn't really have good support for anything related to the Power Query data sources. You can edit the M expressions, but that's it. There's no UI like the one available in SSDT or Power BI.

But more generally, I don't believe Python is available for Analysis Services at all, so it doesn't matter if you're using SSDT or Tabular Editor to build your tabular model. For Power BI models, Python is supported, but [you only a very small number of packages are available](https://docs.microsoft.com/en-us/power-bi/connect-data/service-python-packages-support).

To be completely honest, I have a very hard time understanding why you would want to execute Python scripts in your Tabular model source queries. However, this discussion is totally unrelated to Tabular Editor, so I won't go into more details about that here.

**Comment by akj2784** [comment]
Makes sense. So basically AAS itself doesn't support python, so no question of SSDT and tabular editor supporting python.

**Comment by otykier** [comment]
Correct.
